src/protocolsupport/protocol/typeremapper/itemstack/ItemStackRemapper.java

```diff
@@ -9,13 +9,18 @@
 public class ItemStackRemapper {
 
 	public static NetworkItemStack remapToClient(ProtocolVersion version, String locale,  NetworkItemStack itemstack) {
-		setComplexlyRemapped(itemstack, false);
+		if (version.getProtocolType() != ProtocolType.PE) {
```

Comment:
needs new helper method

Reply:
I will look into this item thingy. Theres still quite some problems here, but it should be fixable within this and complexitemremapper class alone. No need to clone all itemstacks like in entityequipment, because they should be cloned in itemstackserializer.

Reply:
put the cloning back to more how it was. made some misjudgements about where the NetworkItemStack mutations were happening

Reply:
got this cleaned back up, closer to how it was originally. Just cloning in.... 2 spots in the recipes class, and cloning in the inventory transaction stuff